core/drivers/clk/clk-stm32mp25.c

```diff
@@ -0,0 +1,3449 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2024, STMicroelectronics
+ */
+
+#include <config.h>
+#include <drivers/clk_dt.h>
+#include <drivers/stm32_shared_io.h>
+#include <drivers/stm32mp25_rcc.h>
+#include <drivers/stm32mp_dt_bindings.h>
+#include <io.h>
+#include <libfdt.h>
+#include <stdio.h>
+#include <stm32_sysconf.h>
+#include <stm32_util.h>
```

Comment:
Sorry to bother you with that: this file should include
_assert.h_
_kernel/dt.h_
_kernel/panic.h_
_stdbool.h_
_trace.h_
_util.h_


I many (not to say most) cases, we should not rely of the fact a header file we include currently includes another header file we need. This is fine it is relates to definition/declaration done in our header file, but if not, then it looks fine now but may make cause trouble if these header files evolve in the future.